src/haxelib/client/Main.hx

```diff
@@ -954,27 +963,35 @@ class Main {
 		return Path.addTrailingSlash(home) + ".haxelib";
 	}
 
-	function getGlobalRepositoryPath(create = false):String {
+	function getGlobalRepositoriesPath(create = false):String {
 		// first check the env var
-		var rep = Sys.getEnv("HAXELIB_PATH");
-		if (rep != null)
-			return rep.trim();
+		var repEnv = Sys.getEnv("HAXELIB_PATH");
 
 		// try to read from user config
-		rep = try File.getContent(getConfigFile()).trim() catch (_:Dynamic) null;
-		if (rep != null)
-			return rep;
+		var repUser = try File.getContent(getConfigFile()).trim() catch (_:Dynamic) null;
 
+		var repSystem: String = null;
 		if (!IS_WINDOWS) {
 			// on unixes, try to read system-wide config
-			rep = try File.getContent("/etc/.haxelib").trim() catch (_:Dynamic) null;
-			if (rep == null)
-				throw "This is the first time you are runing haxelib. Please run `haxelib setup` first";
+			repSystem = try File.getContent("/etc/.haxelib").trim() catch (_:Dynamic) null;
 		} else {
 			// on windows, try to use haxe installation path
-			rep = getWindowsDefaultGlobalRepositoryPath();
+			repSystem = getWindowsDefaultGlobalRepositoryPath();
 			if (create)
-				try safeDir(rep) catch(e:Dynamic) throw 'Error accessing Haxelib repository: $e';
+				try safeDir(repSystem) catch(e:Dynamic) throw 'Error accessing Haxelib repository: $e';
+		}
+		if (repEnv == null && repUser == null && repSystem == null)
+			throw "This is the first time you are runing haxelib. Please run `haxelib setup` first";
+		var rep: String = "";
+		if (repUser != null)
```

Comment:
The first in the ```reps``` list will be writable repo for ```haxelib install```, so the order matters and that's why ```repUser``` is the first to try - it is the setting of ```haxelib setup```.